### PR TITLE
Provide more detail on how to get the Vagrant Cloud upload to work

### PR DESCRIPTION
**Pull request by jmcgeheeiv**
Thank you for the excellent example.

I had quite a bit of trouble getting this to work with Vagrant Cloud.  The main problem was no error message about my invalid Vagrant Cloud token, for which I even submitted hashicorp/packer#6866.  Wow.

There were a couple more minor struggles, and I documented them for you.

### CODE REVIEW

**Review by azr — APPROVED** [review]
nice one.

File: ubuntu1804.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "variables": {
+    "username": "",
```
Comment (azr) [inline ubuntu1804.json]: when you set `"username": null,` then to quote [usage](https://www.packer.io/docs/templates/user-variables.html#usage) doc.
> If the default value is null, then the user variable will be required. This means that the user must specify a value for this variable or template validation will fail.

this is a nice feature to make sure your users set it by for example running `packer -var username=geerlingguy`

**Comment by jmcgeheeiv** [comment]
As suggested by @azr:
* I changed the user variable defaults to `null` to make them required on the Packer command line
* Added instructions that the variables must be deleted when skipping Vagrant Cloud upload (so they are no longer required on the command line)

**Comment by geerlingguy** [comment]
LGTM. I am going to merge this soon, but I will need to spend a little time updating my other base box configs as well (someday I'll unite them like Bento does...) before merging, just to keep things uniform across all my base box builds.

**Comment by geerlingguy** [comment]
The latest commits make this more of a no-go for me—it's basically a complete rewrite of the documentation, and individually, I am willing to accept some of the changes to clarify bits and pieces of the build process. But as a whole, this makes the README really huge, and also makes this particular README file a lot different than the ones in all my other packer box builds.

Eventually I'm planning on merging them all into one repo ala Bento, but for now, it's hard enough trying to maintain consistent documentation :/

**Comment by jmcgeheeiv** [comment]
Yes, I have battled documentation file skew quite a bit myself, so I understand.

Just merge whatever you like, whenever you like. I will not take offense.

**Comment by geerlingguy** [comment]
This project is deprecated as things are being consolidated into https://github.com/geerlingguy/packer-boxes; see issue: https://github.com/geerlingguy/packer-boxes/issues/45

**Pull request closed without merging**